### PR TITLE
Add DLS in Network Drive

### PR DESCRIPTION
**Pull request by praveen-elastic**
#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

File: connectors/sources/network_drive.py [hunk 1]
```diff
@@ -300,6 +411,112 @@ async def get_content(self, file, timestamp=None, doit=None):
...
+        if file_type == "file":
+            list_permissions = await asyncio.to_thread(
+                self.list_file_permission,
+                file_path=file_path,
+                file_type="file",
+                mode="rb",
+                access=FilePipePrinterAccessMask.READ_CONTROL,
+            )
+        else:
+            list_permissions = await asyncio.to_thread(
```
Comment (timgrein) [inline connectors/sources/network_drive.py:504]: Did you already look into https://github.com/skelsec/aiosmb ? Seems like this is an async wrapper around smb.

**Review by timgrein — COMMENTED** [review]
Left some comments, did you already look into the aiosmb library for Python?

Code looks clean 👏

File: connectors/sources/network_drive.py [hunk 2]
```diff
@@ -152,6 +255,14 @@ def get_default_configuration(cls):
                 "order": 5,
                 "type": "str",
             },
+            "use_document_level_security": {
+                "display": "toggle",
+                "label": "Enable document level security",
+                "order": 6,
+                "tooltip": "Document level security ensures identities and permissions set in Network Drive are maintained in Elasticsearch. This enables you to restrict and personalize read-access users and groups have to documents in this index. Access control syncs ensure this metadata is kept up to date in your Elasticsearch documents.",
```
Comment (leemthompo) [inline connectors/sources/network_drive.py:262]:
```suggestion
                "tooltip": "Document level security ensures identities and permissions set in Network Drive are mirrored in Elasticsearch. This enables you to restrict and personalize read-access users and groups have to documents in this index. Access control syncs ensure this metadata is kept up to date in your Elasticsearch documents.",
```

Comment (leemthompo) [reply connectors/sources/network_drive.py:262]: Looks good with optional word swap :)

File: connectors/sources/network_drive.py [hunk 3]
```diff
@@ -300,6 +411,112 @@ async def get_content(self, file, timestamp=None, doit=None):
...
+                file_path=file_path, file_type=file_type
+            )
+            document[ACCESS_CONTROL] = list(
+                set(document.get(ACCESS_CONTROL, []) + entity_permissions)
+            )
+        return document
+
+    async def _user_access_control_doc(self, user, sid, groups_info, groups_members):
+        prefixed_username = _prefix_user(user)
+        sid_users = _prefix_sid(sid)
```
[the remark this answers is not in the document]
Comment (praveen-elastic) [reply connectors/sources/network_drive.py:448]: You're right, it should be singular, updating it to `sid_user`

File: connectors/sources/network_drive.py (hunk 1 above)
Comment (praveen-elastic) [reply connectors/sources/network_drive.py:504]: Yes, we tried to use the aiosmb module. However it failed to work with MacOS based network drives , so we had to use `smbprotocol`

**Comment by praveen-elastic** [comment]
Buildkite test this

**Comment by praveen-elastic** [comment]
> Left some comments, did you already look into the aiosmb library for Python?
> 
> Code looks clean 👏

Thanks, yes we did look into the aiosmb library, however it failed to work with MacOS network shares, most probably because aiosmb does not support SMB 3, `smbprotocol` did work, so we had to use that.

File: connectors/sources/network_drive.py [hunk 4]
```diff
@@ -93,12 +126,81 @@ async def _remote_validation(self, advanced_rules):
...
+    def session(self):
+        return winrm.Session(
+            self.server_ip,
+            auth=(self.username, self.password),
+            transport="ntlm",
+            server_cert_validation="ignore",
+        )
+
+    def parse_output(self, raw_output):
+        formatted_result = {}
```
[the remark this answers is not in the document]
Comment (praveen-elastic) [reply connectors/sources/network_drive.py:199]: Updated with commit id [4860b1c](https://github.com/elastic/connectors-python/pull/1699/commits/4860b1cdf5a1200de8556ea46307b0a3056d9397)

File: connectors/sources/network_drive.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (praveen-elastic) [reply connectors/sources/network_drive.py]: Updated with commit id [4860b1c](https://github.com/elastic/connectors-python/pull/1699/commits/4860b1cdf5a1200de8556ea46307b0a3056d9397)

File: connectors/sources/network_drive.py [hunk 5]
```diff
@@ -93,12 +126,113 @@ async def _remote_validation(self, advanced_rules):
         )
 
 
+class SecurityInfo:
+    def __init__(self, user, password, server):
+        self.username = user
+        self.server_ip = server
```
Comment (vidok) [inline connectors/sources/network_drive.py:132]: Can this be a host name?

**Review by vidok — APPROVED** [review]

**Pull request merged**